homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -137,6 +138,42 @@ class EnvoyProductionSensorEntityDescription(
 )
 
 
+@dataclass(frozen=True, kw_only=True)
+class EnvoyProductionPhaseSensorEntityDescription(
+    EnvoyProductionSensorEntityDescription
+):
+    """Describes an Envoy production phase sensor entity."""
+
+    on_phase: PhaseNames
+
+
+def phase_sensor_from_production_sensor(
+    sensors: list[EnvoyProductionSensorEntityDescription],
+) -> list[EnvoyProductionPhaseSensorEntityDescription]:
+    """Build phase sensors from production sensors."""
+    return [
+        EnvoyProductionPhaseSensorEntityDescription(
+            key=f"{sensor.key}_l{phase + 1}",
+            translation_key=f"{sensor.translation_key}_phase",
+            native_unit_of_measurement=sensor.native_unit_of_measurement,
+            state_class=sensor.state_class,
+            device_class=sensor.device_class,
+            suggested_unit_of_measurement=sensor.suggested_unit_of_measurement,
+            suggested_display_precision=sensor.suggested_display_precision,
+            value_fn=sensor.value_fn,
+            on_phase=PhaseNames(PHASENAMES[phase]),
+            translation_placeholders={"phase_name": f"l{phase + 1}"},
+        )
```

Comment:
I think `dataclasses.replace` could be used here instead

https://docs.python.org/3/library/dataclasses.html#dataclasses.replace

Reply:
It has an additional item `on_phase` not present in dataclass its getting the data from. Would that work?

Reply:
sadly no. it looks like we are stuck with something complex

Reply:
You might be able to do `dataclasses.asdict` and unpack with `**`

untested... but maybe this will work,

```python
        EnvoyProductionPhaseSensorEntityDescription(
            **dataclasses.asdict(sensor),
            key=f"{sensor.key}_l{phase + 1}",
            translation_key=f"{sensor.translation_key}_phase",
            on_phase=PhaseNames(PHASENAMES[phase]),
            translation_placeholders={"phase_name": f"l{phase + 1}"},
        )
```

Reply:
Doesn't fly, complains about key and translation_key having double values as they are in the asdict list as well. To simplify this, adding the on_phase field to the base class (and not using it) would allow for the dataclasses.replace use.

Reply:
> adding the on_phase field to the base class (and not using it) would allow for the dataclasses.replace use.

That seems like a better idea to me. (typing it to `None` when not used)

Reply:
Done